src/main/java/org/tugraz/sysds/runtime/controlprogram/federated/FederatedWorkerHandler.java

```diff
@@ -64,158 +64,69 @@
 
 public class FederatedWorkerHandler extends ChannelInboundHandlerAdapter {
 	protected static Logger log = Logger.getLogger(FederatedWorkerHandler.class);
-	
+
 	private final IDSequence _seq;
 	private Map<Long, CacheableData<?>> _vars;
-	
+
 	public FederatedWorkerHandler(IDSequence seq, Map<Long, CacheableData<?>> _vars2) {
 		_seq = seq;
 		_vars = _vars2;
 	}
-	
+
 	@Override
 	public void channelRead(ChannelHandlerContext ctx, Object msg) {
-		//TODO could we modularize this method a bit?
 		log.debug("[Federated Worker] Received: " + msg.getClass().getSimpleName());
 		FederatedRequest request;
 		if (msg instanceof FederatedRequest)
 			request = (FederatedRequest) msg;
 		else
 			throw new DMLRuntimeException("FederatedWorkerHandler: Received object no instance of `FederatedRequest`.");
```

Comment:
I know it's not your line of code, but that DMLRuntimeException looks unfitting.

Reply:
Sorry, but it is most fitting here.